src/System.Private.Uri/src/System/UriExt.cs

```diff
@@ -250,7 +248,7 @@ private unsafe bool CheckForEscapedUnreserved(string data)
                         && tempPtr[i + 1] >= '0' && tempPtr[i + 1] <= '7') // max 0x7F
                     {
                         char ch = UriHelper.EscapedAscii(tempPtr[i + 1], tempPtr[i + 2]);
-                        if (ch != c_DummyChar && UriHelper.Is3986Unreserved(ch))
```

Comment:
We named the method using "3986" to remind us that we had modified Uri from the prior [RFC 2396] (https://tools.ietf.org/html/rfc2396).  [RFC 3986](https://tools.ietf.org/html/rfc3986) modified the list of unreserved characters.

So, we should make sure to at least include comments somewhere about what current RFC behavior we support. In .NET Framework, we had to support multiple behaviors and using TFM (Target Framework Moniker) and quirks to support app-compat with legacy targeted apps.

Reply:
Ok.  I'll add a comment, though it's strange that the current code has both Is3986Unreserved and IsUnreserved that are bit-for-bit identical:
https://github.com/dotnet/corefx/blob/97d9fb9d6c7e1d50f527e290ef8e18767dbc6bbf/src/System.Private.Uri/src/System/UriHelper.cs#L711-L726